src/algorithms.jl

```diff
@@ -124,6 +104,8 @@ function BVPSOL(; bvpclass = 2, sol_method = 0, odesolver = nothing)
     return BVPSOL(bvpclass, sol_method, odesolver)
 end
 
+SciMLBase.allows_arbitrary_number_types(alg::BVPSOL) = true
```

Comment:
this is a wrapped method, it's false

Reply:
But without specifying this trait, ```BVPSOL``` would error about using non-standard number type, which I believe is the initial guess that caused this error

```julia
ERROR: Non-standard number type (i.e. not Float32, Float64,
ComplexF32, or ComplexF64) detected as the element type
for the initial condition or time span. These generic
number types are only compatible with the pure Julia
solvers which support generic programming, such as
OrdinaryDiffEq.jl. The chosen solver does not support
this functionality. Please double check that the initial
condition and time span types are correct, and check that
the chosen solver was correct.

Solver: BVPSOL(bvpclass = 2, sol_method = 0)
u0 type: Vector{Vector{Float64}}
Timespan type: Tuple{Float64, Float64}

Some of the types have been truncated in the stacktrace for improved reading. To emit complete information
in the stack trace, evaluate `TruncatedStacktraces.VERBOSE[] = true` and re-run the code.
```

Reply:
OK, I will make the changes in DiffEqBase then